File: hash-table/ransom-note.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        if len(magazine) < len(ransomNote):
            return False

        records = [0]*26

        for char in magazine:
            records[ord(char) - ord('a')]+=1
        for char in ransomNote:
            records[ord(char) - ord('a')]-=1
            if records[ord(char) - ord('a')]<0:
                return False
        return True

File: hash-table/ransom-note.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        if (magazine.size() < ransomNote.size()){
            return false;
        }

        int records[26]={0};
        for (char chara:magazine){
            records[chara-'a']++;
        }

        for (char chara:ransomNote){
            records[chara-'a']--;
            if (records[chara-'a']<0){
                return false;
            }
        }

        return true;
    }
};